packages/theme/src/theme.ts

```diff
@@ -168,20 +170,33 @@ export class Theme extends HTMLElement implements ThemeKindProvider {
     }
 
     protected connectedCallback(): void {
-        /* istanbul ignore if */
-        if (!this.hasAdoptedStyles) {
-            this.adoptStyles();
-        }
+        this.shouldAdoptStyles();
         // Note, first update/render handles styleElement so we only call this if
         // connected after first update.
         /* istanbul ignore if */
         if (window.ShadyCSS !== undefined) {
             window.ShadyCSS.styleElement(this);
         }
+        // Add `this` to the instances array.
+        Theme.instances.push(this);
+    }
+
+    protected disconnectedCallback(): void {
+        const instanceIndex = Theme.instances.indexOf(this);
+        // Remove `this` to the instances array.
+        Theme.instances.splice(instanceIndex, 1);
```

Comment:
Maybe consider just using `Set`, since then you don't need to iterate the index and can just add/delete from the set.